### DISCUSSION TITLE
rustler errors on OTP26

### DISCUSSION THREAD

**Discussion by CharlesOkwuagwu** [rusterlium/rustler#545] [Q&A] [answered]
Please I get this error on OTP26 with rustler 0.28
It's not clear how to apply the upgrade from here 
https://github.com/rusterlium/rustler/blob/master/UPGRADE.md#028---029

```
error: failed to run custom build command for `rustler_sys v2.2.0`

Caused by:
  process didn't exit successfully: `c:/Projects/xler/_build/dev/lib/xler/native/xler_native\release\build\rustler_sys-55353dc99dae555c\build-script-build` (exit code: 101)
  --- stderr
  thread 'main' panicked at 'The NIF version given from RUSTLER_NIF_VERSION is not supported: 2.17', C:\Users\Charles Okwuagwu\.cargo\registry\src\index.crates.io-6f17d22bba15001f\rustler_sys-2.2.0\build.rs:925:9
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...

== Compilation error in file lib/xler/native.ex ==
** (RuntimeError) Rust NIF compile error (rustc exit code 101)
    (rustler 0.28.0) lib/rustler/compiler.ex:41: Rustler.Compiler.compile_crate/2
    lib/xler/native.ex:2: (module)

```

**Comment by filmor** [comment, marked as answer]
For now, there is nothing to do. Just update to 0.29. This problem (having to update for a new NIF version) is gone with 0.29 as well. You'll only have to update if you are actually using new NIF features.

**Reply by CharlesOkwuagwu** [reply]
> For now, there is nothing to do. Just update to 0.29. This problem (having to update for a new NIF version) is gone with 0.29 as well. You'll only have to update if you are actually using new NIF features.

I have followed this instruction. The problem remains.

```
C:\Projects\xler\native\xler_native
λ cargo c
    Updating crates.io index
error: failed to select a version for the requirement `rustler = "^0.29"`
candidate versions found which didn't match: 0.28.0, 0.27.0, 0.26.0, ...
location searched: crates.io index
required by package `xler_native v0.2.0 (C:\Projects\xler\native\xler_native)`
perhaps a crate was updated and forgotten to be re-vendored?

```

**Reply by filmor** [reply]
Ah, sorry, I didn't actually publish that one yet, I'll try to do that this weekend. You can for now either use the current master branch or use the newest rustler_sys version.

**Reply by zanderxyz** [reply]
Hi @filmor, is there an ETA for releasing 0.29?
